core/positions/market_position.go

```diff
@@ -201,3 +201,32 @@ func (p MarketPosition) VWSell() *num.Uint {
 	}
 	return num.UintZero()
 }
+
+func (p MarketPosition) OrderReducesExposure(ord *types.Order) bool {
+	if ord == nil || p.Size() == 0 || ord.PeggedOrder != nil {
+		return false
+	}
+	// long position and short order
+	if p.Size() > 0 && ord.Side == types.SideSell {
+		// market order reduces exposure and doesn't flip position to the other side
+		if p.Size()-int64(ord.Remaining) >= 0 && ord.Type == types.OrderTypeMarket {
+			return true
+		}
+		// sum of all short limit orders wouldn't flip the position if filled (ord already included in pos)
+		if p.Size()-p.Sell() >= 0 && ord.Type == types.OrderTypeLimit {
```

Comment:
This looks a bit off to me: A position of 10 long, with a sell volume on the book of 10, but submitting a sell order for a total volume of 20 would increase the exposure (potential short becomes 20). Shouldn't this check be something like `if p.Size() - (p.Sell() + int64(ord.Remaining)) >= 0`?